### PR TITLE
adds minimal embed focused homepage

### PR DESCRIPTION
**Pull request by npretto**
[[Epic] Meet embedders at the door - minimal edition](https://github.com/metabase/metabase/issues/38233)

### Description

This adds a the minimal homepage from [this design](https://www.figma.com/file/UFay4YjWyUMyl23T1cXzeC/Meet-first-time-embedders-at-the-door-with-a-dashboard?type=design&node-id=1449-15221&mode=design&t=C56e5BXR4bEjeJAk-4).
For the tests, I only modified a current e2e to expect to not find the homepage, and then I made a new flow that selects "Interested in embedding" and expects to find it.

*I will add more tests in a later PR*

### How to verify

1. Do the setup, select that you're interested in embedding or "both"
2. You should see the embedding homepage
3. You should be able to dismiss it and it should go away and never come back again

### Demo

image

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/home/components/EmbedMinimalHomepage/util.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export const showEmbedHomepage = () => {
+  try {
+    localStorage.setItem("showEmbedHomepage", "true");
+  } catch (e) {
+    console.error(e);
```
Comment (npretto) [inline frontend/src/metabase/home/components/EmbedMinimalHomepage/util.ts:5]: localStorage could fail (if disabled, or if in incognito in some browsers).
I still want to see the error if it happens but I don't want to trigger ErrorBoundaries

File: e2e/test/scenarios/onboarding/setup/setup.cy.spec.js [hunk 2]
```diff
@@ -304,6 +309,70 @@ describe("scenarios > setup", () => {
     cy.visit("/browse");
     cy.findByTestId("database-browser").findByText(dbName);
   });
+
+  it("embedded use-case, it should hide the db step and show the embedding homepage", () => {
+    cy.visit("/setup");
+
+    cy.location("pathname").should("eq", "/setup");
```
Comment (npretto) [inline e2e/test/scenarios/onboarding/setup/setup.cy.spec.js:316]: this is a copy and paste of another test, I will refactor them later to extract common bits, for now I wanted to get this merged/have feedback earlier

File: frontend/src/metabase/setup/actions.ts [hunk 3]
```diff
@@ -197,6 +203,13 @@ export const submitSetup = createAsyncThunk<void, void, ThunkConfig>(
         },
       });
 
+      if (usageReason === "embedding" || usageReason === "both") {
+        setShowEmbedHomepageFlag();
+      } else {
+        // make sure that state is clean in case of more than one setup on the same browser
```
Comment (npretto) [inline frontend/src/metabase/setup/actions.ts:209]: Vitorio wasn't able to test the "normal" cases after he triggered the homepage, because he probably didn't dismissed it and the flag stayed in `localStorage` in the following setups.
Multiple setups on the same browser is a pretty rare edge case but the fix is just cleaning the state

File: e2e/test/scenarios/onboarding/setup/setup.cy.spec.js [hunk 4]
```diff
@@ -304,6 +309,70 @@ describe("scenarios > setup", () => {
...
+      cy.findByText("Take me to Metabase").click();
+    });
+
+    cy.location("pathname").should("eq", "/");
+
+    main()
+      .findByText("Get started with Embedding Metabase in your app")
+      .should("exist");
+
+    cy.icon("close").click();
```
Comment (WiNloSt) [inline e2e/test/scenarios/onboarding/setup/setup.cy.spec.js:364]: before we close it can you add a reload here to ensure everything works before you dismiss it?

File: frontend/src/metabase/home/components/HomeContent/HomeContent.tsx [hunk 5]
```diff
@@ -14,9 +15,13 @@ import { HomeRecentSection } from "../HomeRecentSection";
 import { HomeXraySection } from "../HomeXraySection";
 import { getIsXrayEnabled } from "../../selectors";
 import { isWithinWeeks } from "../../utils";
+import { shouldSHowEmbedHomepage } from "../EmbedMinimalHomepage/util";
```
Comment (WiNloSt) [inline frontend/src/metabase/home/components/HomeContent/HomeContent.tsx:18]: I'm not sure this util should be imported from a file under another component. Maybe it would be better to move the util to somewhere else more generic e.g. `frontend/src/metabase/home/utils.ts`.

Also
```suggestion
import { shouldShowEmbedHomepage } from "../EmbedMinimalHomepage/util";
```

File: frontend/src/metabase/home/components/EmbedMinimalHomepage/EmbedMinimalHomepage.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import { NumberedList } from "./EmbedMinimalHomepage.styled";
+
+type EmbedMinimalHomepageProps = {
+  onDismiss: () => void;
+};
+
+export const EmbedMinimalHomepage = ({
+  onDismiss,
+}: EmbedMinimalHomepageProps) => {
+  const learnMoreUrl = useSelector(s =>
```
Comment (WiNloSt) [inline frontend/src/metabase/home/components/EmbedMinimalHomepage/EmbedMinimalHomepage.tsx:17]: Please try to refrain from using abbreviated names, and use the full name for clarity instead.

```suggestion
  const learnMoreUrl = useSelector(state =>
```

File: frontend/src/metabase/home/components/EmbedMinimalHomepage/EmbedMinimalHomepage.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+          <Stack spacing="sm">
+            <Link to="/admin/settings/embedding-in-other-applications">
+              <Button
+                variant="filled"
+                leftIcon={<Icon name="gear" />}
+              >{t`Get started`}</Button>
+            </Link>
+
+            <Text>{jt`${(
+              <ExternalLink href={learnMoreUrl}>{t`Learn more`}</ExternalLink>
```
Comment (WiNloSt) [inline frontend/src/metabase/home/components/EmbedMinimalHomepage/EmbedMinimalHomepage.tsx:66]: Components inside `jt` needs to have `key` because `jt` will convert these to array as far as I understand.

This is the error from this component.
image

**Review by WiNloSt — APPROVED** [review]
I have minor comments, otherwise it looks good to me 👍

File: e2e/test/scenarios/onboarding/setup/setup.cy.spec.js (hunk 4 above)
Comment (npretto) [reply e2e/test/scenarios/onboarding/setup/setup.cy.spec.js:370]: Very good point, added it in 2c75c4253065f42ed9b27b75b089fef0d9098233

File: frontend/src/metabase/home/components/EmbedMinimalHomepage/EmbedMinimalHomepage.tsx (hunk 6 above)
Comment (npretto) [reply frontend/src/metabase/home/components/EmbedMinimalHomepage/EmbedMinimalHomepage.tsx]: 👍 15564908a4185a238c0c1ccdd1d857a34a6501f8

File: frontend/src/metabase/home/components/EmbedMinimalHomepage/EmbedMinimalHomepage.tsx (hunk 7 above)
Comment (npretto) [reply frontend/src/metabase/home/components/EmbedMinimalHomepage/EmbedMinimalHomepage.tsx]: I keep forgetting about it 🤦  bdb803dfc3776a07662b0a6f98db53d16701d51e

File: frontend/src/metabase/home/components/HomeContent/HomeContent.tsx (hunk 5 above)
Comment (npretto) [reply frontend/src/metabase/home/components/HomeContent/HomeContent.tsx]: Moved all the utils to `/home/utils.ts` 👍  e97cde235fa6e0c7b30a95af1bb2999fe6bddee5
